### PR TITLE
ENH: Unique emails for all tickets on order

### LINKED ISSUE #23: Model uniqueness isn't honored in custom Form
Need to revamp the validation to make sure the `unique_together` constraint of our `OrderItem` Model is met on a page that doesn't have access to that information.

### PR DESCRIPTION
**Pull request by thermokarst**
Fixes #23.

Adds custom formset class for multi-form validation.
Also refactors form error template into templatetag.

### CODE REVIEW

File: payments/forms.py [hunk 1]
```diff
@@ -53,3 +53,16 @@ class OrderDetailForm(forms.Form):
...
+    def clean(self):
+        if any(self.errors):
+            return
+        emails = []
+        for form in self.forms:
+            email = form.cleaned_data['email']
+            if email in emails:
+                raise forms.ValidationError('Tickets must have a unique '
+                                            'email address.')
+            emails.append(email)
```
Comment (jakereps) [inline payments/forms.py]: Does this need to be calling `super` and returning the `cleaned_data` like the other form that had the cleaning hooked into?

**Comment by jakereps** [comment]
:+1: Just one thing.

File: payments/forms.py (hunk 1 above)
Comment (thermokarst) [inline payments/forms.py]: I don't think so, `clean` is just a [hook](https://github.com/django/django/blob/stable/1.10.x/django/forms/formsets.py#L365), but I can add a note or do the noop `super` for documentation purposes.

Comment (jakereps) [inline payments/forms.py]: Oh weird, figured it worked just like all the other ones that actually did something, but was on my phone so couldn't test it or look into it very easily. Guess this just lets you get at the data that was already cleaned by the individual forms `clean` methods.

**Pull request merged**